src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/EngineGeneralTest.java

```diff
@@ -1989,6 +1992,109 @@ public void betweenTest() throws DatabaseEngineException {
         );
     }
 
+    @Test
+    public void testWith() throws DatabaseEngineException {
+        if (this.engine instanceof MySqlEngine) {
```

Comment:
This is not cool given that this is an abstract test.  
I thought this class was abstract and each engine would extend it but I have just realize that this is not true.

That said, I suggest not comparing `this.engine` but `config.engine` because it is a string an you may remove the dependency.
But again, this is better but still not cool. :/